ansible_builder/steps.py

```diff
@@ -46,7 +46,21 @@ def __init__(self, requirements_naming):
             "RUN ansible-galaxy role install -r /build/{0} --roles-path {1}".format(
                 requirements_naming, constants.base_roles_path),
             "RUN ansible-galaxy collection install -r /build/{0} --collections-path {1}".format(
-                requirements_naming, constants.base_collections_path)
+                requirements_naming, constants.base_collections_path),
+            "",
+            "RUN mkdir -p {0} {1}".format(constants.base_roles_path, constants.base_collections_path),
+        ])
+
+
+class GalaxyCopySteps(Steps):
+    def __init__(self):
+        """Assumes given requirements file name has been placed in the build context
+        """
+        self.steps = []
+        self.steps.extend([
+            "",
+            "COPY --from=builder {0} {0}".format(constants.base_roles_path),
+            "COPY --from=builder {0} {0}".format(constants.base_collections_path),
```

Comment:
What is we copied in to the base image (from builder) the tarball folder of collections.   Then, we could run ansible-galaxy install tarballs/*.tar.gz again.

The main idea here is, there will be uscases where uses will build custom collections tarballs, and need to install them over something from the galaxy server. And people could remove the need for requirement files in the builder phase, and copy in their own tarballs.

This is kinda of how we are doing it with python wheels today in python-builder / python-base images: https://github.com/ansible/python-builder-image/ https://github.com/ansible/python-base-image/

Reply:
I like this idea, but we may want to do this in a followup PR depending on whether @beeankha thinks she can accomplish this before she leaves for winter break. I think she was hoping to get something merged before then.